packages/gestalt/src/Image.js

```diff
@@ -29,133 +32,109 @@ type Props = {|
 /**
  * https://gestalt.pinterest.systems/Image
  */
-export default class Image extends PureComponent<Props> {
-  static propTypes = {
-    alt: PropTypes.string.isRequired,
-    children: PropTypes.node,
-    color: PropTypes.string,
-    elementTiming: PropTypes.string,
-    fit: PropTypes.oneOf(['contain', 'cover', 'none']),
-    importance: PropTypes.oneOf(['high', 'low', 'auto']),
-    loading: PropTypes.oneOf(['eager', 'lazy', 'auto']),
-    naturalHeight: PropTypes.number.isRequired,
-    naturalWidth: PropTypes.number.isRequired,
-    onError: PropTypes.func,
-    onLoad: PropTypes.func,
-    role: PropTypes.oneOf(['img', 'presentation']),
-    sizes: PropTypes.string,
-    src: PropTypes.string.isRequired,
-    srcSet: PropTypes.string,
-  };
-
-  static defaultProps: {|
-    color: string,
-    fit?: 'contain' | 'cover' | 'none',
-    importance?: 'high' | 'low' | 'auto',
-    loading?: 'eager' | 'lazy' | 'auto',
-  |} = {
-    color: 'transparent',
-    fit: 'none',
-    importance: 'auto',
-    loading: 'auto',
-  };
+export default function Image({
+  alt,
+  children,
+  color = 'transparent',
+  elementTiming,
+  fit = 'none',
+  importance = 'auto',
+  loading = 'auto',
+  naturalHeight,
+  naturalWidth,
+  onError,
+  onLoad,
+  role = 'img',
+  sizes,
+  src,
+  srcSet,
+}: Props): Node {
+  const isScaledImage = shouldScaleImage(fit);
 
-  componentDidMount() {
-    if (shouldScaleImage(this.props.fit)) {
-      this.loadImage();
-    }
-  }
+  const handleLoad = useCallback(() => {
+    onLoad?.();
+  }, [onLoad]);
 
-  componentDidUpdate(prevProps: Props) {
-    const { fit, src } = this.props;
-    if (shouldScaleImage(fit) && prevProps.src !== src) {
-      this.loadImage();
-    }
-  }
+  const handleError = useCallback(() => {
+    onError?.();
+  }, [onError]);
 
-  handleLoad: () => void = () => {
-    if (this.props.onLoad) {
-      this.props.onLoad();
-    }
-  };
-
-  handleError: () => void = () => {
-    if (this.props.onError) {
-      this.props.onError();
-    }
-  };
-
-  loadImage() {
+  const loadImage = useCallback(() => {
     if (typeof window !== 'undefined') {
       const image = new window.Image();
-      image.onload = this.handleLoad;
-      image.onerror = this.handleError;
-      image.src = this.props.src;
+      image.onload = handleLoad;
+      image.onerror = handleError;
+      image.src = src;
```

Comment:
I think    

` if (shouldScaleImage(fit) && prevProps.src !== src) {
`
is not equivalent to passing `src` to the dep array.

Maybe we could save src in a `useRef` and check it against new values (then update useRef)